pycbc/workflow/core.py

```diff
@@ -764,6 +764,98 @@ def _filename(self, ifo, description, extension, segment):
         return "%s-%s-%s-%s.%s" % (ifo, description.upper(), start, 
                                    duration, extension)  
     
+class RemoteFile(File):
+    '''
+    This class holds the details of an individual output file 
+    This file(s) may be pre-supplied, generated from within the workflow
+    command line script, or generated within the workflow. The important stuff
+    is:
+
+    * The ifo that the File is valid for
+    * The time span that the OutFile is valid for
+    * A short description of what the file is
+    * The extension that the file should have
+    * The url where the file should be located
+
+    An example of initiating this class:
+    
+    c = File("H1", "INSPIRAL_S6LOWMASS", segments.segment(815901601, 815902001), file_url="file://localhost/home/spxiwh/H1-INSPIRAL_S6LOWMASS-815901601-400.xml.gz" )
+
+    another where the file url is generated from the inputs:
+
+    c = File("H1", "INSPIRAL_S6LOWMASS", segments.segment(815901601, 815902001), directory="/home/spxiwh", extension="xml.gz" )
+    '''
+    def __init__(self, ifos, exe_name, segs, pegasus_lfn,
+                 directory=None, tags=None, 
+                 store_file=True, use_tmp_subdirs=False):
+        """
```

Comment:
I don't think pegasus_lfn is a good name, just use lfn or the full logical_filename. It should also be optional. If not given, it should use the standard behavior of constructing a filename.